src/Runner.Worker/JobRunner.cs

```diff
@@ -209,6 +210,53 @@ private async Task<TaskResult> CompleteJobAsync(IJobServer jobServer, IExecution
             jobContext.Debug($"Finishing: {message.JobDisplayName}");
             TaskResult result = jobContext.Complete(taskResult);
 
+            if (_runnerSettings.DisableUpdate == true)
+            {
+                try
+                {
+                    var currnetVersion = new PackageVersion(BuildConstants.RunnerPackage.Version);
+                    ServiceEndpoint systemConnection = message.Resources.Endpoints.Single(x => string.Equals(x.Name, WellKnownServiceEndpointNames.SystemVssConnection, StringComparison.OrdinalIgnoreCase));
+                    VssCredentials serverCredential = VssUtil.GetVssCredential(systemConnection);
+
+                    var runnerServer = HostContext.GetService<IRunnerServer>();
+                    await runnerServer.ConnectAsync(systemConnection.Url, serverCredential);
+                    var serverPackages = await runnerServer.GetPackagesAsync("agent", BuildConstants.RunnerPackage.PackageName, 5, false, CancellationToken.None);
+                    if (serverPackages.Count > 0)
+                    {
+                        serverPackages = serverPackages.OrderByDescending(x => x.Version).ToList();
+                        Trace.Info($"Newer packages {StringUtil.ConvertToJson(serverPackages.Select(x => x.Version.ToString()))}");
+
+                        var warnOnFailedJob = false; // any minor/patch version behind.
+                        var warnOnOldRunnerVersion = false; // >= 2 minor version behind
+                        if (serverPackages.Any(x => x.Version.CompareTo(currnetVersion) > 0))
+                        {
+                            Trace.Info($"Current runner version {currnetVersion} is behind the latest runner version {serverPackages[0].Version}.");
+                            warnOnFailedJob = true;
+                        }
+
+                        if (serverPackages.Where(x => x.Version.Minor > currnetVersion.Minor).Count() > 1)
+                        {
+                            Trace.Info($"Current runner version {currnetVersion} is way behind the latest runner version {serverPackages[0].Version}.");
+                            warnOnOldRunnerVersion = true;
+                        }
+
+                        if (result == TaskResult.Failed && warnOnFailedJob)
+                        {
+                            jobContext.Warning($"The job failure might caused by an out of date runner (current runner version is {currnetVersion}). Please update runner to latest version {serverPackages[0].Version} {serverPackages[0].DownloadUrl}.");
+                        }
+                        else if (warnOnOldRunnerVersion)
+                        {
+                            jobContext.Warning($"The current runner version is {currnetVersion}. Please update runner to latest version {serverPackages[0].Version} {serverPackages[0].DownloadUrl}.");
```

Comment:
@ethomson this is the warning that need to be tweak

Reply:
I like it - I would tweak it a bit to align with what we're planning to add in the service (@fhammerl).

> You are currently using runner version {currentVersion}. This version is out of date.  Please update to the latest version {serverPackages[0].Version} at  {serverPackages[0].DownloadUrl}");

Reply:
fixed.